deltacat/compute/compactor_v2/utils/compaction_utils.py

```diff
@@ -0,0 +1,713 @@
+            file_index,
+            file_index + mat_result.pyarrow_write_result.files,
+        )
+
+        file_index += mat_result.pyarrow_write_result.files
+        previous_task_index = mat_result.task_index
+
+    s3_utils.upload(
+        mutable_compaction_audit.audit_url,
+        str(json.dumps(mutable_compaction_audit)),
+        **params.s3_client_kwargs,
+    )
+
+    deltas: List[Delta] = [m.delta for m in mat_results]
+
+    # Note: An appropriate last stream position must be set
+    # to avoid correctness issue.
+    merged_delta: Delta = Delta.merge_deltas(
+        deltas,
+        stream_position=params.last_stream_position_to_compact,
+    )
+
+    return merged_delta, mat_results, hb_id_to_entry_indices_range
+
+
+def _get_compaction_result(
+    params: CompactPartitionParams,
+    mutable_compaction_audit,
+    compacted_partition,
+    round_completion_info,
+    audit_url,
+    hb_id_to_entry_indices_range,
+    rcf_source_partition_locator,
+    new_compacted_delta_locator,
+    pyarrow_write_result,
+) -> ExecutionCompactionResult:
+
+    input_inflation = None
+    input_average_record_size_bytes = None
+    # Note: we only consider inflation for incremental delta
+    if (
+        mutable_compaction_audit.input_size_bytes
+        and mutable_compaction_audit.hash_bucket_processed_size_bytes
+    ):
+        input_inflation = (
+            mutable_compaction_audit.hash_bucket_processed_size_bytes
+            / mutable_compaction_audit.input_size_bytes
+        )
+
+    if (
+        mutable_compaction_audit.hash_bucket_processed_size_bytes
+        and mutable_compaction_audit.input_records
+    ):
+        input_average_record_size_bytes = (
+            mutable_compaction_audit.hash_bucket_processed_size_bytes
+            / mutable_compaction_audit.input_records
+        )
+
+    logger.info(
+        f"The inflation of input deltas={input_inflation}"
+        f" and average record size={input_average_record_size_bytes}"
+    )
+
+    # After all incremental delta related calculations, we update
+    # the input sizes to accommodate the compacted table
+    if round_completion_info:
+        mutable_compaction_audit.set_input_file_count(
+            (mutable_compaction_audit.input_file_count or 0)
+            + round_completion_info.compacted_pyarrow_write_result.files
+        )
+        mutable_compaction_audit.set_input_size_bytes(
+            (mutable_compaction_audit.input_size_bytes or 0.0)
+            + round_completion_info.compacted_pyarrow_write_result.file_bytes
+        )
+        mutable_compaction_audit.set_input_records(
+            (mutable_compaction_audit.input_records or 0)
+            + round_completion_info.compacted_pyarrow_write_result.records
+        )
+
+    s3_utils.upload(
+        mutable_compaction_audit.audit_url,
+        str(json.dumps(mutable_compaction_audit)),
+        **params.s3_client_kwargs,
+    )
+
+    new_round_completion_info = RoundCompletionInfo.of(
```

Comment:
We can put this remaining function into a new function like `write_new_round_completion_file`?